devdoc/buffer_requirements.md

```diff
@@ -39,7 +39,7 @@ extern BUFFER_HANDLE BUFFER_create(const unsigned char* source, size_t size);
 
 BUFFER_create creates a new buffer from the memory at source, having size "size".
 **SRS_BUFFER_02_001: [**If source is NULL then BUFFER_create shall return NULL.**]** 
-**SRS_BUFFER_02_002: [**Otherwise, BUFFER_create shall allocate memory to hold size bytes and shall copy from source size bytes into the newly allocated memory.**]** 
+**SRS_BUFFER_02_002: [**Otherwise, BUFFER_create shall allocate memory to hold size bytes and shall copy from source size bytes into the newly allocated memory. If size parameter is 0 then buffer field of BUFFER_TAG shall still hold a valid pointer whose content is '\0' and size field shall be 0.**]** 
```

Comment:
Can you please make this a separate requirement ID (SRS_BUFFER_99_001)?
The reason we try to keep these requirements slim is so we can trace that we implemented the tests for them.
Each test is tagged with at least a requirement ID and we also tag this in the code...

We have a tool that runs as part of the gated checkin we have that can check the coverage so we know requirements are covered.

---
In reply to: [75431413](https://github.com/Azure/azure-c-shared-utility/pull/20#discussion_r75431413) [](ancestors = 75431413)